controllers/storagecluster/cephcluster.go

```diff
@@ -1178,16 +1178,39 @@ func createPrometheusRules(r *StorageClusterReconciler, sc *ocsv1.StorageCluster
 		return err
 	}
 	applyLabels(getCephClusterMonitoringLabels(*sc), &prometheusRule.ObjectMeta)
-	replaceTokens := []exprReplaceToken{
+
+	replaceTokens := []replaceToken{
 		{
 			recordOrAlertName: "CephMgrIsAbsent",
-			wordInExpr:        "openshift-storage",
+			wordToReplace:     "openshift-storage",
 			replaceWith:       sc.Namespace,
 		},
 	}
+
+	// if different than default nearFullRatio or fullRatio values are provided, replace the values in the prometheus rule accordingly
+	nearFullRatio := *getFullRatios(sc, "nearFull")
+	fullRatio := *getFullRatios(sc, "full")
+	criticallyFullRatio := (nearFullRatio + fullRatio) / 2
+
+	if nearFullRatio != defaults.DefaultNearFullRatio {
+		replaceTokens = append(replaceTokens,
+			createReplaceToken("", "", "75%", fmt.Sprintf("%.2f%%", nearFullRatio*100)),
+			createReplaceToken("", "", "0.75", fmt.Sprintf("%f", nearFullRatio)))
+	}
+	if criticallyFullRatio != (defaults.DefaultNearFullRatio+defaults.DefaultFullRatio)/2 {
```

Comment:
@travisn My assumption is criticallyFullRatio is something that's being used in the alerts & it's a mid point between nearFullRatio & FullRatio. Am I correct?

Or should this value be coming from the backFillFull ratio?

Reply:
Thanks for the clarification, updated.